src/Lib.P2P/PeerConnection.cs

```diff
@@ -0,0 +1,433 @@
+            {
+                // eat it.
+            }
+
+            _log.Debug($"stop reading messsages from {RemoteAddress}");
+        }
+
+        /// <summary>
+        ///   Starts reading messages from the remote peer on the specified stream.
+        /// </summary>
+        internal async Task ReadMessagesAsync(Stream stream, CancellationToken cancel)
+        {
+            IPeerProtocol protocol = new Multistream1();
+            try
+            {
+                while (!cancel.IsCancellationRequested && stream != null && stream.CanRead)
+                {
+                    await protocol.ProcessMessageAsync(this, stream, cancel).ConfigureAwait(false);
+                }
+            }
+            catch (EndOfStreamException)
+            {
+                // eat it.
+            }
+            catch (Exception e)
+            {
+                if (!cancel.IsCancellationRequested && stream != null)
+                {
+                    _log.Error($"reading message failed {RemoteAddress} {RemotePeer}", e);
+                }
+            }
+        }
+
+        #region IDisposable Support
+
+        private bool _disposedValue; // To detect redundant calls
+
+        /// <summary>
+        ///   Signals that the connection is closed (disposed).
+        /// </summary>
+        public event EventHandler<PeerConnection> Closed;
+
+        /// <summary>
+        ///  TODO
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposedValue)
+            {
+                return;
+            }
+            
+            _disposedValue = true;
+
+            if (!disposing)
+            {
+                return;
+            }
+            
+            _log.Debug($"Closing connection to {RemoteAddress}");
+            if (Stream != null)
+            {
+                try
+                {
+                    Stream.Dispose();
+                }
+                catch (ObjectDisposedException)
+                {
+                    // ignore stream already closed.
+                }
+                catch (Exception e)
+                {
+                    _log.Warn($"Failed to close connection to {RemoteAddress}", e);
+
+                    // eat it.
+                }
+                finally
+                {
+                    Stream = null;
+                    _statsStream = null;
+                }
+            }
+
+            SecurityEstablished.TrySetCanceled();
+            IdentityEstablished.TrySetCanceled();
+            IdentityEstablished.TrySetCanceled();
+            Closed?.Invoke(this, this);
+
+            // free unmanaged resources (unmanaged objects) and override a finalizer below.
+            // set large fields to null.
+        }
+
+        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+        // ~PeerConnection() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)